jdaviz/core/template_mixin.py

```diff
@@ -1403,6 +1601,13 @@ def _selected_changed(self, event):
                     return
         return super()._selected_changed(event)
 
+    def add_filter(self, *filters):
+        super().add_filter(*filters)
+        if 'reference_has_wcs' in filters:
+            # reference data can change whenever data is added OR removed from a viewer
+            self.hub.subscribe(self, AddDataMessage, handler=self._on_viewers_changed)
+            self.hub.subscribe(self, RemoveDataMessage, handler=self._on_viewers_changed)
```

Comment:
@bmorris3 - this should be updated if there is an event for changing reference data in #2179